### PR TITLE
Makes backendRatelimit response code configurable

### LINKED ISSUE #1762: Make HTTP response code of backendRatelimit configurable
Currently other rate limiting filters support this feature.

For example, docs of `ratelimit`:

```
Parameters:

    number of allowed requests per time period (int)
    time period for requests being counted (time.Duration)
    response status code to use for a rate limited request - optional, default: 429

ratelimit(20, "1m")
ratelimit(300, "1h")
ratelimit(4000, "1m", 503)
```

Please implement the same for `backendRatelimit` for consistency.

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Fixes #1762

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
👍

File: filters/ratelimit/backendratelimit.go [hunk 1]
```diff
@@ -35,19 +38,29 @@ func (*backendRatelimit) CreateFilter(args []interface{}) (filters.Filter, error
...
+		f.StatusCode = code
+	}
+	return f, nil
 }
 
-func (f *backendRatelimit) Request(ctx filters.FilterContext) {
+func (limit *BackendRatelimit) Request(ctx filters.FilterContext) {
 	// allows overwrite
-	ctx.StateBag()[filters.BackendRatelimit] = f.settings
+	ctx.StateBag()[filters.BackendRatelimit] = limit
```
Comment (AlexanderYastrebov) [inline filters/ratelimit/backendratelimit.go:63]:
```
$ go test -run NONE -bench Benchmark -benchmem
goos: linux
goarch: amd64
pkg: github.com/zalando/skipper/filters/ratelimit
cpu: Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
BenchmarkStateBagValue-8        12864667                87.65 ns/op           80 B/op          1 allocs/op
BenchmarkStateBagPointer-8      62434342                16.09 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/zalando/skipper/filters/ratelimit    2.256s
```

<details>

```go
func BenchmarkStateBagValue(b *testing.B) {
	limit := testBackendRatelimit()
	x := limit
	stateBag := make(map[string]interface{})

	for i := 0; i < b.N; i++ {
		stateBag[filters.BackendRatelimit] = x
		x = stateBag[filters.BackendRatelimit].(BackendRatelimit)
	}
}

func BenchmarkStateBagPointer(b *testing.B) {
	limit := testBackendRatelimit()
	x := &limit
	stateBag := make(map[string]interface{})

	for i := 0; i < b.N; i++ {
		stateBag[filters.BackendRatelimit] = x
		x = stateBag[filters.BackendRatelimit].(*BackendRatelimit)
	}
}

func testBackendRatelimit() BackendRatelimit {
	return BackendRatelimit{
		Settings: ratelimit.Settings{
			Type:       ratelimit.ClusterServiceRatelimit,
			Group:      "backend.group",
			MaxHits:    10,
			TimeWindow: time.Second,
		},
		StatusCode: http.StatusServiceUnavailable,
	}
}
```
</details>

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**